apps/theming-designer/src/components/SemanticSlots.tsx

```diff
@@ -432,33 +58,107 @@ export const SemanticSlots: React.StatelessComponent<ISemanticSlotsProps> = (pro
     );
   };
 
-  const fillVariantSlotsList = (variantType: VariantThemeType, mappingType: {}): JSX.Element[] => {
+  const trimPaletteSlots = (paletteSlots: IPalette): IPaletteSlots => {
+    let trimmedPaletteSlots: IPaletteSlots = {};
+    for (let palette in paletteSlots) {
+      if (
+        palette.startsWith('theme') ||
+        palette.startsWith('neutral') ||
+        palette === 'black' ||
+        palette === 'white' ||
+        palette === 'redDark' ||
+        palette === 'accent'
+      ) {
+        trimmedPaletteSlots[palette] = (paletteSlots as any)[palette];
+      }
+    }
+    return trimmedPaletteSlots;
+  };
+
+  const isASemanticColor = (semanticString: string): boolean => {
+    return (
+      semanticString.startsWith('body') ||
+      semanticString.startsWith('primary') ||
+      semanticString.startsWith('disabled') ||
+      semanticString.startsWith('focus') ||
+      semanticString.startsWith('variant') ||
+      semanticString.startsWith('default') ||
+      semanticString.includes('action') ||
+      semanticString.includes('link') ||
+      semanticString.startsWith('accent') ||
+      semanticString.startsWith('list') ||
+      semanticString.startsWith('button') ||
+      semanticString.startsWith('menu') ||
+      semanticString.startsWith('input') ||
+      semanticString.startsWith('small')
+    );
+  };
+
+  const trimSemanticSlotsOrNames = <T extends unknown>(semanticSlots: T) => {
```

Comment:
Consider using type overloads instead of doing what you're currently doing, which would be simpler. The current thing is.. equivalent to just saying `(semanticSlots: any) => { }` 

Hard for me to prescribe the exact syntax to use as it isn't super obvious what the possible inputs are and what the output in each case will be (maybe this is a sign that this function can use JSDoc or some inline comments)

Reply:
Oh I just realized you actually did add comments.. (which did not show in the old iteration) but it doesn't really explain input/outputs.